src/tests/ftest/control/dmg_telemetry_net.py

```diff
@@ -0,0 +1,158 @@
+          replication.
+          Stop one server, let rebuild start and complete.
+          Destroy container and create a new one.
+          Use mdtest to create 120K files of size 32K with 3-way
+          replication.
+          Stop one more server in the middle of mdtest. Let rebuild to complete.
+          Allow mdtest to complete.
+          Destroy container and create a new one.
+          Use mdtest to create 120K files of size 32K with 3-way
+          replication.
+          Stop 2 servers in the middle of mdtest. Let rebuild to complete.
+          Allow mdtest to complete.
+
+        :avocado: tags=all,pr,daily_regression
+        :avocado: tags=hw,medium
+        :avocado: tags=control,telemetry,net
+        :avocado: tags=test_with_telemetry_net,test_net_telemetry
+        """
+
+        metrics = "engine_net_ofi_sockets_req_timeout"
+        data = self.telemetry.get_metrics(metrics)
+
+        req_timeouts = 0
+        for host in data:
+            for metric in data[host]["engine_net_ofi_sockets_req_timeout"]["metrics"]:
+                if metric["value"] > 0:
+                    req_timeouts += metric["value"]
+
+        # set params
+        targets = self.params.get("targets", "/run/server_config/*")
+        rank = self.params.get("rank_to_kill", "/run/testparams/*")
+        self.dmg = self.get_dmg_command()
+
+        # create pool
+        self.add_pool(connect=False)
+
+        # make sure pool looks good before we start
+        checks = {
+            "pi_nnodes": len(self.hostlist_servers),
+            "pi_ntargets": len(self.hostlist_servers) * targets,
+            "pi_ndisabled": 0,
+        }
+        self.assertTrue(
+            self.pool.check_pool_info(**checks),
+            "Invalid pool information detected before rebuild")
+
+        self.assertTrue(
+            self.pool.check_rebuild_status(rs_errno=0, rs_done=1,
+                                           rs_obj_nr=0, rs_rec_nr=0),
+            "Invalid pool rebuild info detected before rebuild")
+
+        # create 1st container
+        self.add_container(self.pool)
+        # start 1st mdtest run and let it complete
+        self.execute_mdtest()
+        # Kill rank[6] and wait for rebuild to complete
+
+        self.server_managers[0].stop_ranks([rank[0]], self.d_log, force=True)
+
+        #########################################################
+        #
+        # START: Check engine_net_req_timeout telemetry values
+        #
+        #########################################################
+
+        # Remove the killed host from the clustershell telemetry hostlist
+        self.telemetry.hosts.remove(self.hostlist_servers[rank[0]])
+
+        data = self.telemetry.get_metrics(metrics)
+
+        req_timeouts = 0
+        for host in data:
+            for metric in data[host]["engine_net_ofi_sockets_req_timeout"]["metrics"]:
+                req_timeouts += metric["value"]
+
+        if not req_timeouts > 0:
```

Comment:
(pylint-unneeded-not) Consider changing "not req_timeouts > 0" to "req_timeouts <= 0"